src/dotnet/APIView/docs/apiview-lightweight-permissions-proposal.md

```diff
@@ -0,0 +1,277 @@
+{
+    userName = userName ?? User.GetGitHubLogin();
+    var userProfile = await _userProfileCache.GetUserProfileAsync(userName);
+    
+    // Compute effective permissions from group memberships
+    userProfile.Permissions = await _permissionsManager.GetEffectivePermissionsAsync(userName);
+    
+    return new LeanJsonResult(userProfile, StatusCodes.Status200OK);
+}
+```
+
+Define the permission types:
+
+```typescript
+// Global roles (no language scope)
+export type GlobalRole = "Unknown" | "ServiceTeam" | "SdkTeam" | "Admin";
+
+// Language-scoped roles (require language)
+export type LanguageScopedRole = "DeputyArchitect" | "Architect";
+
+// Discriminated union for role assignments
+export type RoleAssignment = 
+    | { kind: "global"; role: GlobalRole }
+    | { kind: "scoped"; role: LanguageScopedRole; language: string };
+
+export interface EffectivePermissions {
+    userId: string;
+    roles: RoleAssignment[];
+}
+```
+
+### Permission Helper Service
+
+Create a service to check permissions in the UI:
+
+```typescript
+@Injectable({ providedIn: 'root' })
+export class PermissionsService {
+    
+    /** Check if user has a global role */
+    hasGlobalRole(permissions: EffectivePermissions, role: GlobalRole): boolean {
+        return permissions.roles.some(r => 
+            r.kind === "global" && r.role === role
+        );
+    }
+    
+    /** Check if user has a language-scoped role for a specific language */
+    hasLanguageRole(permissions: EffectivePermissions, role: LanguageScopedRole, language: string): boolean {
+        return permissions.roles.some(r => 
+            r.kind === "scoped" && r.role === role && r.language === language
+        );
+    }
+    
+    /** Check if user is Admin */
+    isAdmin(permissions: EffectivePermissions): boolean {
+        return this.hasGlobalRole(permissions, "Admin");
+    }
+    
+    /** Check if user can approve for a specific language */
+    canApprove(permissions: EffectivePermissions, language: string): boolean {
+        return this.isAdmin(permissions) ||
+               this.hasLanguageRole(permissions, "Architect", language) ||
+               this.hasLanguageRole(permissions, "DeputyArchitect", language);
+    }
+}
+```
+
+### Example Response
+
+`GET /api/userprofile` response:
+
+```json
+{
+  "userName": "almend",
+  "email": "almend@microsoft.com",
+  "preferences": { ... },
+  "permissions": {
+    "userId": "almend",
+    "roles": [
+      { "kind": "global", "role": "SdkTeam" },
+      { "kind": "scoped", "role": "Architect", "language": "Python" },
+      { "kind": "scoped", "role": "DeputyArchitect", "language": "Java" }
+    ]
+  }
+}
+```
+
+### UI Usage Example
+
+```typescript
+// In a component
+if (this.permissionsService.canApprove(this.userProfile.permissions, review.language)) {
+    this.showApproveButton = true;
+}
+
+if (this.permissionsService.isAdmin(this.userProfile.permissions)) {
```

Comment:
I still prefer just making `hasGlobalRole` accept an array because otherwise, every other conditional is gonna need an `isAdmin` tagged on... OR... I guess we could just assign the admins the `SDKTeam` role as well 🤔